users/tests/test_views.py

```diff
@@ -496,3 +496,56 @@ def test_menu_expansion_update_should_update_menu_expansion_setting_in_user_pref
 
         user_menu_expansion_setting_after_update = CustomUserPreferences.objects.get(user=self.employee).expanded_menu
         self.assertEqual(user_menu_expansion_setting_after_update, not expanded_menu)
+
+
+class ChangePasswordOnLoginTests(TestCase):
+    def setUp(self):
+        self.user_password = "password"
+        self.user = UserFactory(need_change_password=True)
+
+    def test_change_user_password_on_login_view_should_change_user_password_on_post(self):
+        data = {"old_password": self.user_password, "new_password1": "newuserpasswd", "new_password2": "newuserpasswd"}
+        self.client.login(email=self.user.email, password=self.user_password)
+        response = self.client.post(reverse("password_change_on_login"), data)
+        self.user.refresh_from_db()
+        self.assertEqual(response.status_code, 302)
+        self.assertTrue(self.user.check_password(data["new_password1"]))
+
+    def test_change_user_password_on_login_view_should_not_change_user_password_when_old_password_is_incorrect(self):
+        data = {"old_password": "wronguserpasswd", "new_password1": "newuserpasswd", "new_password2": "newuserpasswd"}
+        self.client.login(email=self.user.email, password=self.user_password)
+        response = self.client.post(reverse("password_change_on_login"), data)
+        self.user.refresh_from_db()
+        self.assertEqual(response.status_code, 200)
+        self.assertFalse(self.user.check_password(data["new_password1"]))
+        self.assertFalse(self.user.check_password(data["old_password"]))
+        self.assertTrue(self.user.check_password(self.user_password))
+
+    def test_change_user_password_on_login_view_should_not_change_user_password_when_new_passwords_does_not_match(self):
+        data = {
+            "old_password": self.user_password,
+            "new_password1": "newuserpasswd",
+            "new_password2": "notthesamenewuserpasswd",
+        }
+        self.client.login(email=self.user.email, password=self.user_password)
+        response = self.client.post(reverse("password_change_on_login"), data)
+        self.user.refresh_from_db()
+        self.assertEqual(response.status_code, 200)
+        # Replacement between "’" and "'" it is caused because in response
+        # we got "’" but from strings hardcoded in `strings.py` we got "'"
+        self.assertEqual(
+            response.context["form"].errors.get("new_password2")[0].replace("’", "'"),
+            ValidationErrorText.VALIDATION_ERROR_SIGNUP_PASSWORD_MESSAGE,
+        )
+        self.assertFalse(self.user.check_password(data["new_password1"]))
+        self.assertTrue(self.user.check_password(data["old_password"]))
+
+    def test_change_user_password_on_login_view_should_clear_need_change_password_flag_after_successful_password_change(
```

Comment:
Please also here split test using spaces to logical parts

Reply:
Ok.